go-controller/pkg/util/pod_annotation.go

```diff
@@ -468,9 +468,34 @@ func AddRoutesGatewayIP(
 		podAnnotation.Gateways = append(podAnnotation.Gateways, network.GatewayRequest...)
 		topoType := netinfo.TopologyType()
 		switch topoType {
-		case types.Layer2Topology, types.LocalnetTopology:
+		case types.LocalnetTopology:
 			// no route needed for directly connected subnets
 			return nil
+		case types.Layer2Topology:
+			if !IsNetworkSegmentationSupportEnabled() || !netinfo.IsPrimaryNetwork() {
+				return nil
+			}
+			for _, podIfAddr := range podAnnotation.IPs {
+				isIPv6 := utilnet.IsIPv6CIDR(podIfAddr)
+				nodeSubnet, err := MatchFirstIPNetFamily(isIPv6, nodeSubnets)
+				if err != nil {
+					return err
+				}
+				gatewayIPnet := GetNodeGatewayIfAddr(nodeSubnet)
```

Comment:
For L2 at IC we should point to .1 address instead of node gateway, as proposed on the enhancement topology.

Reply:
yes that is correct, the nodeSubnet on L2 will be same on all nodes its actually the flat pod subnet, so this works.